frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -378,7 +378,8 @@ export const PublicFormProvider = ({
   const [submissionData, setSubmissionData] = useState<SubmissionData>()
 
   const {
-    data,
+    // WARNING: This is the latest form definition which should not be used for >= 2nd step of MRF which should use snapshotted form definition.
+    data: latestNonMrfConsistentFormData,
```

Comment:
we could call this `latestNonMrfConsistentFormData`, although I'm wondering if something shorter might be more readable (e.g. `latestFormdata`, with the MRF issue explained in the comments)!
<img width="839" height="232" alt="image" src="https://github.com/user-attachments/assets/634a90f5-4f39-45ad-86da-f80bb0cb0817" />